mcs/mcs/expression.cs

```diff
@@ -1733,11 +1766,20 @@ void EmitLoad (EmitContext ec)
 
 					Variable.EmitAssign (ec);
 
-					if (expr_unwrap != null) {
-						ec.MarkLabel (no_value_label);
-					} else if (!value_on_stack) {
+					if (!vtype_variable)
 						Variable.Emit (ec);
-					}
+				}
+
+				if (vtype_variable) {
+					var end = ec.DefineLabel ();
+
+					ec.EmitInt (1);
+					ec.Emit (OpCodes.Br_S, end);
+					ec.MarkLabel (nonmatching_label);
+					ec.EmitInt (0);
+					ec.MarkLabel (end);
+				} else {
+					ec.MarkLabel (nonmatching_label);
 				}
```

Comment:
I don't understand why this special casing for `vtype_variable` is needed. can't we always emit `dup` and bail out to the `nonmatching_label` at the very end?

Reply:
good find, it's probably slower but not that much